### DISCUSSION TITLE
How do i increase draw distance of the editor?

### DISCUSSION THREAD

**Discussion by SaleeshAbduSaleem** [stride3d/stride#2209] [Newcomers] [answered]
![Screenshot (134)](https://github.com/stride3d/stride/assets/148552677/3d61647c-eeed-408b-a41a-36e36943b323)

**Comment by Doprez** [comment]
~~Doesnt seem to be an option currently, you can do it for the in game camera but I cant see any options for the camera in the veiwport config. by default it should be 0.1 to 1,000 units.~~

~~It may be handy to add that option to the below settings window in the future.~~

**Nicogo has the answer below.**

Now for some "fun" workarounds if you desperate to have this right now, you can use EntityProcessors to accomplish this on a per scene basis.

WARNING: EntityProcessors are great but can cause some weird issues when run in editor(often crashes), so back up and save your projects frequently.

First you will want to create an EntityComponent just as a way to tell the editor that a script needs to be run in the editor.
CameraSettingsComponent.cs
``` csharp
[DataContract(nameof(CameraSettingsComponent))]
[ComponentCategory("Utils")]
[DefaultEntityComponentProcessor(typeof(CameraSettingsProcessor))]
public class CameraSettingsComponent : EntityComponent
{
	/// <summary>
	/// Forces the editor to update the camera clip planes.
	/// </summary>
	[DataMemberIgnore]
	public bool ValuesChanged { get; set; }

	public float NearPlane
	{
		get => _nearPlane;
		set
		{
			_nearPlane = value;
			ValuesChanged = true;
		}
	}
	private float _nearPlane = 0.1f;

	public float FarPlane
	{
		get => _farPlane;
		set
		{
			_farPlane = value;
			ValuesChanged = true;
		}
	}
	private float _farPlane = 1000;
}
```

Then create the processor that will actually update the camera settings.
CameraSettingsProcessor.cs
``` csharp
public class CameraSettingsProcessor : EntityProcessor<CameraSettingsComponent>
{
	public override void Update(GameTime time)
	{

		var sceneSystem = Services.GetService<SceneSystem>();
		if (sceneSystem == null)
			return;

		foreach (var pair in ComponentDatas)
		{
			var component = pair.Key;
			var data = pair.Value;

			if (component.ValuesChanged)
			{
				var camera = TryGetMainCamera(sceneSystem);
				camera.NearClipPlane = component.NearPlane;
				camera.FarClipPlane = component.FarPlane;

				component.ValuesChanged = false;
			}
		}
	}

	public CameraComponent TryGetMainCamera(SceneSystem sceneSystem)
	{
		CameraComponent camera = null;
		if (sceneSystem.GraphicsCompositor.Cameras.Count == 0)
		{
			// The compositor wont have any cameras attached if the game is running in editor mode
			// Search through the scene systems until the camera entity is found
			// This is what you might call "A Hack"
			foreach (var system in sceneSystem.Game.GameSystems)
			{
				if (system is SceneSystem editorSceneSystem)
				{
					foreach (var entity in editorSceneSystem.SceneInstance.RootScene.Entities)
					{
						if (entity.Name == "Camera Editor Entity")
						{
							camera = entity.Get<CameraComponent>();
							break;
						}
					}
				}
			}
		}
		else
		{
			camera = sceneSystem.GraphicsCompositor.Cameras[0].Camera;
		}

		return camera;
	}
}
```

Finally add the component to an entity in the scene and you now have a settings file you can update as needed :)
![image](https://github.com/stride3d/stride/assets/73259914/4cd295ef-7b9f-4bba-8daf-5b65535a6644)

**Comment by Nicogo1705** [comment, marked as answer]
you can set camera near/far plane there : 

![image](https://github.com/stride3d/stride/assets/20603105/a358eac2-7e12-4f19-93ee-9abcba30ce25)

**Reply by Doprez** [reply]
Well that is SO much easier than my suggestion lol.
